pymc3/gp/gp.py

```diff
@@ -3,60 +3,117 @@
 
+
+def GP(name, X, mean_func=None, cov_func=None,
+       cov_func_noise=None, sigma2=None,
+       approx=None, n_inducing=None, inducing_points=None,
+       observed=None, chol_const=True, *args, **kwargs):
+    """Gausian process constructor
     Parameters
     ----------
+    X : array
+        Grid of points to evaluate Gaussian process over.
     mean_func : Mean
         Mean function of Gaussian process
     cov_func : Covariance
         Covariance function of Gaussian process
-    X : array
-        Grid of points to evaluate Gaussian process over. Only required if the
-        GP is not an observed variable.
-    sigma : scalar or array
-        Observation standard deviation (defaults to zero)
+    cov_func_noise : Covariance
+        Covariance function of noise process (ignored for approximations)
+    sigma2 : scalar or array
+        Observation variance
+    approx : None or string,
+        Allowed values are 'FITC' and 'VFE'
+    n_inducing : integer
+        The number of inducing points to use
+    inducing_points : array
+        The inducing points to use
+    observed : None or array
+        The observed 'y' values, use if likelihood is Gaussian
+    chol_const: boolean
+        Whether or not to stabilize Cholesky decompositions
     """
-    def __init__(self, mean_func=None, cov_func=None, X=None, sigma=0, *args, **kwargs):
+    CHOL_CONST = chol_const
 
-        if mean_func is None:
-            self.M = Zero()
-        else:
-            if not isinstance(mean_func, Mean):
-                raise ValueError('mean_func must be a subclass of Mean')
-            self.M = mean_func
-
-        if cov_func is None:
-            raise ValueError('A covariance function must be specified for GPP')
-        if not isinstance(cov_func, Covariance):
-            raise ValueError('cov_func must be a subclass of Covariance')
-        self.K = cov_func
+    if mean_func is None:
+        mean_func = Zero()
+    else:
+        if not isinstance(mean_func, Mean):
+            raise ValueError('mean_func must be a subclass of Mean')
+    if cov_func is None:
+        raise ValueError('A covariance function must be specified for GPP')
+    if not isinstance(cov_func, Covariance):
+        raise ValueError('cov_func must be a subclass of Covariance')
+
+    ##### NONCONJUGATE
+    if observed is None:
+        gp = GPFullNonConjugate(name, X, mean_func, cov_func)
+        return gp.RV
 
-        self.sigma = sigma
+    ##### CONJUGATE
+    if approx is None:
+        if sigma2 is None and cov_func_noise is None:
+            raise ValueError('Provide a value or a prior for the noise variance')
+        if sigma2 is not None and cov_func_noise is None:
+            cov_func_noise = lambda X: sigma2 * tt.eye(X.shape[0])
+        return GPFullConjugate(name, X, mean_func, cov_func, cov_func_noise, observed=observed)
+    else:
+        approx = approx.upper()
+
+    ##### CONJUGATE, APPROXIMATION
+    if approx not in ["VFE", "FITC"]:
```

Comment:
What happens if inducing points are passed but approx is None? Should you perhaps just set one as the default?